lizmap/modules/lizmap/classes/lizmapLogItem.class.php

```diff
@@ -54,18 +54,21 @@ class lizmapLogItem
         'email',
     );
 
-    public function __construct($key)
+    public function __construct($key, $readConfigPath)
     {
-        // read the lizmap log configuration file
-        $readConfigPath = parse_ini_file(jApp::varPath().$this->config, true);
-
         $section = 'item:'.$key;
 
         // Check if this item exists in the ini file
         if (array_key_exists($section, $readConfigPath)) {
             // Set each property
             foreach (self::$properties as $property) {
-                $this->data[$property] = $readConfigPath[$section][$property];
+                if (isset($readConfigPath[$section][$property])) {
+                    $this->data[$property] = $readConfigPath[$section][$property];
+                }
+                // not sure about those lines, are every properties essential ? if yes uncomment
```

Comment:
every properties are not essential. `getData` can return `null` if the property is not defined in `$this->data`.